contracts/gateway/EIP20Gateway.sol

```diff
@@ -505,17 +505,14 @@ contract EIP20Gateway is GatewayBase {
     {
         require(
             _messageHash != bytes32(0),
-            "Message hash must not be zero"
+            "Message hash must not be zero."
         );
 
-        // get the message object for the _messageHash
         MessageBus.Message storage message = messages[_messageHash];
 
-        require(message.sender == msg.sender, "Only staker can revert stake.");
-
         require(
-            message.intentHash != bytes32(0),
```

Comment:
Why is this deleted?

Reply:
Because that was duplicate, this check `        require(message.sender == msg.sender, "Only staker can revert stake.");	` is sufficient to validate if message exists.

Reply:
In that case do you require the following?
```solidity
        require(
            _messageHash != bytes32(0),
            "Message hash must not be zero"
        );
```

Reply:
`bytes32(0)` is valid hash but we will not allow `bytes32(0)` hash since it can be due to invalid input data. We can discuss this if we want to allow it.